packages/flutter_tools/lib/src/ios/plist_parser.dart

```diff
@@ -71,9 +151,9 @@ class PlistParser {
   /// If [key] is not found in the property list, this will return null.
   ///
   /// The [plistFilePath] and [key] arguments must not be null.
-  String? getValueFromFile(String plistFilePath, String key) {
+  dynamic getValueFromFile(String plistFilePath, String key) {
```

Comment:
This handles all the valid plist xml types, but is not used except in `getStringValueFromFile`.  Will it be used for something other than strings in g3?  If not, we could get rid of this method and make `getStringValueFromFile` call:
```
plutil -extract $key raw -expect string $plistFilePath
```
instead of parsing the whole file.  And we can implement other types as needed.

Reply:
> Will it be used for something other than strings in g3?

No. My intention is only to replace `plutil` (mac only) with `plistutil` (runs on linux).

> If not, we could get rid of this method and make getStringValueFromFile call:
> plutil -extract $key raw -expect string $plistFilePath
> instead of parsing the whole file.

Getting rid of this method SGTM.

About calling `plutil -extract` for `getStringValueFromFile`, we already need to call `plutil -convert` for `parseFile`. I personally think it would be a better idea to consolidate the call to external tool in a single place, and plist files are generally small and parsing it is usually fast.

I've removed `getValueFromFile` and changed `getStringValueFromFile` to return `parseFile(plistFilePath)[key] as String?`. Do let me know if you still think we should use `plutil -extract` for `getStringValueFromFile`.

Thanks!

Reply:
You're right, it's a cheap operation.  Let's keep it simple for the g3 case.